apps/cowswap-frontend/src/legacy/state/price/hooks.ts

```diff
@@ -26,16 +25,8 @@ type SetQuoteErrorCallback = (params: SetQuoteErrorParams) => void
 
 type QuoteParams = { chainId?: ChainId; token?: string | null }
 
-export const useAllQuotes = ({
-  chainId,
-}: Partial<Pick<ClearQuoteParams, 'chainId'>>): Partial<QuotesMap> | undefined => {
-  return useSelector<AppState, Partial<QuotesMap> | undefined>((state) => {
-    const quotes = chainId && state.price.quotes[chainId]
-
-    if (!quotes) return {}
-
-    return quotes
-  })
+export const useAllQuotes = (chainId: SupportedChainId): Partial<QuotesMap> | undefined => {
+  return useSelector<AppState, Partial<QuotesMap> | undefined>((state) => state.price.quotes[chainId])
```

Comment:
```
Selector unknown returned a different result when called with the same parameters. This can lead to unnecessary rerenders.
Selectors that return a new reference (such as an object or an array) should be memoized: https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization 
```